cloud-controller-manager/do/kubernetes_lb_validator.go

```diff
@@ -0,0 +1,181 @@
+		if currentLBID == "" {
+			v.Log.Info("missing kubernetes.digitalocean.com/load-balancer-id annotation")
+			resp, err = v.validateCreate(ctx, lbRequest)
+			if err != nil {
+				if v.checkError(resp) {
+					v.Log.Error(err, "failed to validate lb creation")
+					return admission.Denied(fmt.Sprintf("failed to validate lb creation: %v", err))
+				}
+				v.Log.Error(err, "failed to validate lb creation")
+				return admission.Errored(int32(resp.StatusCode), err)
+			}
+			return admission.Allowed("can not validate lb without annotation: kubernetes.digitalocean.com/load-balancer-id annotation")
+		}
+
+		// perform update if associated with created lb
+		if (len(svc.Status.LoadBalancer.Ingress) > 0 && svc.Status.LoadBalancer.Ingress[0].IP != "") || currentLBID != "" {
+			v.Log.Info(fmt.Sprintf("updating lb id: %v", currentLBID))
+			resp, err = v.validateUpdate(ctx, currentLBID, lbRequest)
+			if err != nil {
+				if v.checkError(resp) {
+					v.Log.Error(err, "failed to validate lb update")
+					return admission.Denied(fmt.Sprintf("failed to validate lb update: %v", err))
+				}
+				v.Log.Error(err, "failed to validate lb update")
+				return admission.Errored(int32(resp.StatusCode), err)
+			}
+			v.Log.Info("lb update validated")
+			return admission.Allowed("valid update request")
+		}
+	}
+
+	// validate create request otherwise
+	v.Log.Info("validating create request")
+	resp, err = v.validateCreate(ctx, lbRequest)
+	if err != nil {
+		if v.checkError(resp) {
+			v.Log.Error(err, "failed to validate lb creation")
+			return admission.Denied(fmt.Sprintf("failed to validate lb creation: %v", err))
+		}
+		v.Log.Error(err, "failed to validate lb creation")
+		return admission.Errored(int32(resp.StatusCode), err)
+	}
+	v.Log.Info(fmt.Sprintf("allowing create"))
+	return admission.Allowed("valid lb create request")
+}
+
+func (v *KubernetesLBServiceValidator) validateCreate(ctx context.Context, lbRequest *godo.LoadBalancerRequest) (*godo.Response, error) {
+	_, resp, err := v.GClient.LoadBalancers.Create(ctx, lbRequest)
+	return resp, err
+}
+
+func (v *KubernetesLBServiceValidator) validateUpdate(ctx context.Context, currentLBID string, lbRequest *godo.LoadBalancerRequest) (*godo.Response, error) {
+	_, resp, err := v.GClient.LoadBalancers.Update(ctx, currentLBID, lbRequest)
+	return resp, err
+}
+
+func (v *KubernetesLBServiceValidator) buildRequest(name string, region string, forwardingRules []godo.ForwardingRule) *godo.LoadBalancerRequest {
+	return &godo.LoadBalancerRequest{
+		Name:            name,
+		Tag:             "",
+		Region:          region,
+		ForwardingRules: forwardingRules,
+		ValidateOnly:    true,
+	}
+}
+
+func (v *KubernetesLBServiceValidator) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+func (v *KubernetesLBServiceValidator) checkError(response *godo.Response) bool {
+	if response.StatusCode == http.StatusUnprocessableEntity {
```

Comment:
This will cause a nil pointer dereference if `response` is nil, which is the case whenever we hit a network error (i.e., we don't even get an HTTP response). We'll need to check for that.

Reply:
Please address this comment.